File: backend/src/main/java/com/roadsaathi/backend/dto/HeatmapResponse.java
package com.roadsaathi.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HeatmapResponse {

    @Builder.Default
    private String type = "FeatureCollection";

    private List<FeatureDto> features;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class FeatureDto {
        @Builder.Default
        private String type = "Feature";
        private GeometryDto geometry;
        private PropertiesDto properties;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class GeometryDto {
        @Builder.Default
        private String type = "Point";
        private List<Double> coordinates;
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class PropertiesDto {
        private String clusterId;
        private String hazardType;
        private Integer count;
        private String aiBrief;
        private Integer severity;
        private String nhCorridor;
    }
}

File: android/app/src/main/java/com/roadsaathi/data/remote/dto/HeatmapResponse.kt
package com.roadsaathi.data.remote.dto

import com.google.gson.annotations.SerializedName

data class HeatmapResponse(
    @SerializedName("type")
    val type: String = "FeatureCollection",
    @SerializedName("features")
    val features: List<HeatmapFeature>
)

data class HeatmapFeature(
    @SerializedName("type")
    val type: String = "Feature",
    @SerializedName("geometry")
    val geometry: HeatmapGeometry,
    @SerializedName("properties")
    val properties: HeatmapProperties
)

data class HeatmapGeometry(
    @SerializedName("type")
    val type: String = "Point",
    @SerializedName("coordinates")
    val coordinates: List<Double>
)

data class HeatmapProperties(
    @SerializedName("cluster_id")
    val clusterId: String,
    @SerializedName("hazard_type")
    val hazardType: String,
    @SerializedName("count")
    val count: Int,
    @SerializedName("ai_brief")
    val aiBrief: String? = null,
    @SerializedName("severity")
    val severity: Int = 1,
    @SerializedName("nh_corridor")
    val nhCorridor: String? = null
)
